viewer/event_db.cpp

```diff
@@ -1,38 +1,122 @@
 #include "event_db.hpp"
 
+#include <algorithm>
+#include <queue>
+
 namespace HawkTracer
 {
 namespace viewer
 {
 
 EventDB::EventDB()
 {
-
 }
 
 void EventDB::insert(parser::Event event)
 {
-    _events.push_back(std::move(event));
+
+    auto klass_id = event.get_klass()->get_id();
+    if (_events.find(klass_id) == _events.end())
+    {
+        _events[klass_id] = std::vector<parser::Event>(1, std::move(event));
+    }
+    else
+    {
+        std::vector<parser::Event>::reverse_iterator it;
+        for (it = _events[klass_id].rbegin(); it != _events[klass_id].rend(); ++it)
+        {   
+            if ((*it).get_timestamp() < event.get_timestamp())
```

Comment:
You can get rid of this explicit dereference; instead of `(*it).get_timestamp()`, you can write `it->get_timestamp()`